api_client.py

```diff
@@ -209,8 +210,10 @@ def request_url(self, host_url, body, header, retry_count=3):
                 raise
             return response.read()
 
-    def get_alerts_or_events(self):
+    def get_alerts_or_events(self, write_fnc):
```

Comment:
This refactoring to pass a function object in seems unnecessary and obscures the code.